alerts/machine-config-operator/MachineConfigControllerPausedPoolKubeletCA.md

```diff
@@ -0,0 +1,155 @@
+  trust it.
+
+Aside from these symptoms, your nodes should still be okay until
+`kube-apiserver-to-kubelet-signer` expires.
+
+### Long term: critical
+
+The `kube-apiserver-to-kubelet-signer` certificate in the `kubelet-ca.crt`
+bundle is what enables trust between your node's kubelet and your cluster. The
+apiserver schedules a rotation when this certificate reaches 80% of its
+(currently 365 day) lifetime. The new one *must* be deployed before all of the
+previous ones expire.
+
+The kubelets on the nodes in the specified pool will stop communicating with the
+cluster apiserver if the `kube-apiserver-to-kubelet-signer`  is allowed to
+expire. The nodes will no longer be able to participate in the cluster. *This is
+very bad*.
+
+To avoid this, you will need to unpause the specified pool to allow the new
+`kubelet-ca.crt` bundle to be deployed.
+
+## Diagnosis
+
+If a pool is paused, it was paused by an administrator; pools do not pause
+themselves automatically.
+
+For the commands below, replace `$MCP_NAME` with the name of your pool.
+
+You can see the pool's paused status by looking at the `spec.paused` field for
+the pool:
+
+```console
+oc -n openshift-machine-config-operator get mcp $MCP_NAME -o jsonpath='{.spec.paused}'
+```
+
+For the `kube-apiserver-to-kubelet-signer` certificate in the cluster, you can
+check its annotations to see when it was rotated:
+
+```console
+oc -n openshift-kube-apiserver-operator describe secret kube-apiserver-to-kubelet-signer
+```
+
+The `kubelet-client` cert (the one that is responsible for `oc logs`, etc
+working) in the cluster, you can check which signer it was signed with:
+
+```console
+oc -n openshift-kube-apiserver describe secret/kubelet-client
+```
+
+You can also check the certificates that are present in the bundle on one of
+your nodes to see when they expire:
+
+```console
+openssl crl2pkcs7 -nocrl -certfile /etc/kubernetes/kubelet-ca.crt | openssl pkcs7 -print_certs -text -noout
+``````
+
+>NOTE: There may be multiple `kube-apiserver-to-kubelet-signer` certificates in
+>the certificate bundles, kube-apiserver does not pull them out until they
+>expire. You want to look at the *newest* `kube-apiserver-to-kubelet-signer`.
+
+You can find the MachineConfig that a pool is currently using by looking at the
+pool status:
+
+```console
+oc -n openshift-machine-config-operator get mcp $MCP_NAME -o jsonpath='{.status.configuration.name}'
+```
+
+Checking the expiry dates in the MachineConfig yourself is less pleasant because
+the bundle is encoded, but you can use something like this (you have to
+urldecode the ignition file contents, assumes you have python2):
+
+```console
+oc get mc rendered-worker-f318c7173a1785bc3bf846c558a2ad49 -o jsonpath='{.spec.config.storage.files[?(@.path=="/etc/kubernetes/kubelet-ca.crt")].contents.source}'  | python2 -c "import urllib; print(urllib.unquote(raw_input()))" | openssl x509 -text -noout
```

Comment:
nit: Python 2 is [end-of-life upstream][1] (I dunno what the state is for RHEL).  Can we use Python 3?  Testing against CI assets:

```console
$ curl -s https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/periodic-ci-openshift-release-master-nightly-4.10-e2e-aws-serial/1506535707059949568/artifacts/e2e-aws-serial/gather-extra/artifacts/machineconfigs.json | jq -r '.items[] | select(.metadata.name | startswith("rendered-worker-")).spec.config.storage.files[] | select(.path == "/etc/kubernetes/kubelet-ca.crt").contents.source' | python3 -c 'import sys, urllib.parse; print(urllib.parse.unquote(sys.stdin.read()))' | openssl x509 -text -noout
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number: 383882405144258650 (0x553d3037a25685a)
        Signature Algorithm: sha256WithRSAEncryption
...
```

[1]: https://peps.python.org/pep-0373/#maintenance-releases

Reply:
Absolutely we can use Python3. Thank you. I did waver on that and leaned toward what I thought was "lowest common denominator". ( I feel old now. Python 3 isn't "new" anymore :laughing: ) 

Aware that machineconfig extraction UX is not the greatest. Long term I have some code I would like to get into `oc` to extract a file from a machineconfig so users don't have mess with inline python scripts and ridiculous jsonpath expressions.